### ISSUE TITLE
Name Clash for Actions since 0.36 (support function overloads)

### ISSUE THREAD

**Issue by tutenkum** [odata2ts/odata2ts#314]
I've tried to upgrade from 0.35 to 0.37 and had issues with compile errors of the generated code.

The odata model we have to use has issues with duplicate definitions of actions and functions. The functions itself won't be an issue since they are handled properly. However the actions will now generate duplicates, since they are no longer renamed.

After I take a look into the code I have an idea what is going wrong. Before the change name clashes were prevented in the  [`NameClashValidator`](https://github.com/odata2ts/odata2ts/blob/520f6fee95b7c5616ce163dd11851b5de15a8d04/packages/odata2ts/src/data-model/validation/NameClashValidator.ts) for actions and functions by renaming them if necessary. Now this check isn't present anymore to support overrides of functions. They are now just added to the data model regardless the name clash. While handling the functions properly in the [`DataModel`](https://github.com/odata2ts/odata2ts/blob/520f6fee95b7c5616ce163dd11851b5de15a8d04/packages/odata2ts/src/data-model/DataModel.ts) methods [`addUnboundOperationType()`](https://github.com/odata2ts/odata2ts/blob/520f6fee95b7c5616ce163dd11851b5de15a8d04/packages/odata2ts/src/data-model/DataModel.ts#L197) and [`addBoundOperationType()`](https://github.com/odata2ts/odata2ts/blob/520f6fee95b7c5616ce163dd11851b5de15a8d04/packages/odata2ts/src/data-model/DataModel.ts#L219) the actions are not handled nor checked for name clashes anymore.

Regardless the issues within our odata model the behavior of the generator has changed and the output is no longer compiling. I would expect at least a warning regarding the name clashes for actions. However I would prefer the same handling as before to be functional compliant to the previous versions.

**Comment by texttechne** [comment]
Hi @tutenkum,

so your use case is a faulty EDMX model, right?
Along these lines?
```
      <Action Name="DoStuff" IsBound="true">
        <Parameter Name="_it" Type="SRV.WarehouseType"
                   Nullable="false"/>
      </Action>
      <!-- Name Clash -->
      <Action Name="DoStuff" IsBound="true">
        <Parameter Name="_it" Type="SRV.WarehouseType"
                   Nullable="false"/>
      </Action>
```
At least I can reproduce your name clash resolution issue with this example...

While this can be fixed for actions, it cannot be fixed this for functions while also supporting function overloads. As you found out. This would mean that you cannot call those duplicated functions. And here comes my question: Are you just ignoring those duplicated functions? You don't use them?

Is there any background why you have this duplications in your EDMX?

**Comment by texttechne** [comment]
Ok, thanks for the detailed answer & the background. I'm always interested in that kind of stuff.

Then it's fixed with [@odata2ts/odata2ts@0.38.0](https://github.com/odata2ts/odata2ts/releases/tag/%40odata2ts%2Fodata2ts%400.38.0)

Please test & close the issue if the test was successful.
